### ISSUE TITLE
Installation issue: libvterm and neovim

### ISSUE THREAD

**Issue by G-Ragghianti** [LLNL/spack#38219]
### Steps to reproduce the issue

```console
$ spack install neovim
wmv2wit  neovim@0.8.3%gcc@12.2.1~ipo~no_luajit build_system=cmake build_type=Release generator=make arch=linux-fedora37-skylake
wnvphgc      ^cmake@3.26.3%gcc@12.2.1~doc+ncurses+ownlibs~qt build_system=generic build_type=Release arch=linux-fedora37-skylake
ooy7vsn          ^ncurses@6.4%gcc@12.2.1~symlinks+termlib abi=none build_system=autotools arch=linux-fedora37-skylake
gqqc2dq          ^openssl@1.1.1t%gcc@12.2.1~docs~shared build_system=generic certs=mozilla arch=linux-fedora37-skylake
kfect2g              ^ca-certificates-mozilla@2023-01-10%gcc@12.2.1 build_system=generic arch=linux-fedora37-skylake
b2ynhqc              ^zlib@1.2.13%gcc@12.2.1+optimize+pic+shared build_system=makefile arch=linux-fedora37-skylake
md25h3f      ^gettext@0.21.1%gcc@12.2.1+bzip2+curses+git~libunistring+libxml2+tar+xz build_system=autotools arch=linux-fedora37-skylake
dk4mxab          ^bzip2@1.0.8%gcc@12.2.1~debug~pic+shared build_system=generic arch=linux-fedora37-skylake
v5pqkc3              ^diffutils@3.9%gcc@12.2.1 build_system=autotools arch=linux-fedora37-skylake
wzijbui          ^libxml2@2.10.3%gcc@12.2.1~python build_system=autotools arch=linux-fedora37-skylake
vxlsz2w          ^tar@1.34%gcc@12.2.1 build_system=autotools zip=pigz arch=linux-fedora37-skylake
mxotrjw              ^pigz@2.7%gcc@12.2.1 build_system=makefile arch=linux-fedora37-skylake
y3fqyfv              ^zstd@1.5.5%gcc@12.2.1+programs build_system=makefile compression=none libs=shared,static arch=linux-fedora37-skylake
nyudywx          ^xz@5.4.1%gcc@12.2.1~pic build_system=autotools libs=shared,static arch=linux-fedora37-skylake
ery72xb      ^gmake@4.4.1%gcc@12.2.1~guile build_system=autotools arch=linux-fedora37-skylake
jdl2til      ^gperf@3.1%gcc@12.2.1 build_system=autotools arch=linux-fedora37-skylake
5mtcp27      ^jemalloc@5.3.0%gcc@12.2.1~debug~documentation+fill~prof~stats build_system=autotools jemalloc_prefix=none libs=shared,static arch=linux-fedora37-skylake
g7tztjf      ^libiconv@1.17%gcc@12.2.1 build_system=autotools libs=shared,static arch=linux-fedora37-skylake
7rvb2gc      ^libluv@1.44.2-1%gcc@12.2.1~ipo build_system=cmake build_type=Release generator=make arch=linux-fedora37-skylake
j7swsuh      ^libtermkey@0.22%gcc@12.2.1 build_system=makefile arch=linux-fedora37-skylake
2el7t4c          ^gzip@1.12%gcc@12.2.1 build_system=autotools arch=linux-fedora37-skylake
jc5l3pv          ^libtool@2.4.7%gcc@12.2.1 build_system=autotools arch=linux-fedora37-skylake
bjbwsbi              ^m4@1.4.19%gcc@12.2.1+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-fedora37-skylake
3niym4o                  ^libsigsegv@2.14%gcc@12.2.1 build_system=autotools arch=linux-fedora37-skylake
c5oydr7      ^libuv@1.44.1%gcc@12.2.1 build_system=autotools arch=linux-fedora37-skylake
egws55r      ^libvterm@0.3.1%gcc@12.2.1 build_system=makefile arch=linux-fedora37-skylake
2yufva3      ^lua-lpeg@1.0.2-1%gcc@12.2.1 build_system=lua arch=linux-fedora37-skylake
26nhhe6      ^lua-luajit-openresty@2.1-20220111%gcc@12.2.1+lualinks build_system=makefile fetcher=curl arch=linux-fedora37-skylake
ye6ses7          ^curl@8.0.1%gcc@12.2.1~gssapi~ldap~libidn2~librtmp~libssh~libssh2~nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-fedora37-skylake
66665t4          ^unzip@6.0%gcc@12.2.1 build_system=makefile arch=linux-fedora37-skylake
ox4cje4      ^lua-mpack@1.0.9%gcc@12.2.1 build_system=lua arch=linux-fedora37-skylake
2nhxfjb      ^msgpack-c@3.1.1%gcc@12.2.1~ipo build_system=cmake build_type=Release generator=make arch=linux-fedora37-skylake
dalxgzx      ^pkgconf@1.9.5%gcc@12.2.1 build_system=autotools arch=linux-fedora37-skylake
jc4sp44      ^tree-sitter@0.20.8%gcc@12.2.1 build_system=makefile arch=linux-fedora37-skylake
fknsp65      ^unibilium@2.0.0%gcc@12.2.1 build_system=generic arch=linux-fedora37-skylake
zgrgjll          ^perl@5.36.0%gcc@12.2.1~cpanm+open+shared+threads build_system=generic arch=linux-fedora37-skylake

```

### Error message

<details><summary>Error message</summary>

<pre>
...
==> Installing libvterm-0.3.1-egws55rawm2b6wvvcgz4vrr2gdnakewo
==> No binary for libvterm-0.3.1-egws55rawm2b6wvvcgz4vrr2gdnakewo found: installing from source
==> Error: FetchError: All fetchers failed for spack-stage-libvterm-0.3.1-egws55rawm2b6wvvcgz4vrr2gdnakewo
==> Warning: Skipping build of neovim-0.8.3-wmv2witunleovndk3dqshot5s2smvnuy since libvterm-0.3.1-egws55rawm2b6wvvcgz4vrr2gdnakewo failed
</pre></details>

### Information on your system

* **Spack:** 0.21.0.dev0 (4c2531d5fbc2022dd144d1c59dd6faa7ce2c9607)
* **Python:** 3.11.3
* **Platform:** linux-fedora37-skylake
* **Concretizer:** clingo

### Additional information

The fetchers fail to pull the source tar file for libvterm from the origin host because the host is flagged by Fortinet as a malicious site.  I think this is likely why the spack source cache also doesn't contain the tar file. Is there anything that can be done to host this source on a more reputable site?

@albestro  @trws

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by albestro** [comment]
I quickly checked my remindings and `neovim` project allows to use a bundled `libvterm`.

Looking at https://github.com/neovim/neovim/blob/master/cmake.deps/deps.txt
the bundled `libvterm` is downloaded from https://github.com/neovim/deps repository.

I think it might be better to let `neovim` build the bundled `libvterm`. Otherwise, we can add a version pointing to the `neovim` maintained repository with deps directly in `libvterm`.

Currently the only spack package known to use `libvterm` is `neovim`. Probably it is not worth anymore keep maintaining the separate `libvterm` package and let `neovim` take care of it? That's why I would strongly prefer the former option.

**Comment by albestro** [comment]
Ok, counter-order. `neovim` build system is not very friendly with me and it does not look like it would easily allow to build bundled dependencies. There is an internal subproject `cmake.deps`, that would take care of building dependencies, but it is not fully embedded in the main build (as far as I can tell).

The workaround of using the repository maintained by `neovim` might still be valid somehow, even if it is not purely clean.

```patch
diff --git a/var/spack/repos/builtin/packages/libvterm/package.py b/var/spack/repos/builtin/packages/libvterm/package.py
index 0954967bd6..811bb5e0f5 100644
--- a/var/spack/repos/builtin/packages/libvterm/package.py
+++ b/var/spack/repos/builtin/packages/libvterm/package.py
@@ -13,6 +13,11 @@ class Libvterm(MakefilePackage):
     homepage = "http://www.leonerd.org.uk/code/libvterm/"
     url = "http://www.leonerd.org.uk/code/libvterm/libvterm-0.1.3.tar.gz"

+    version(
+        "0.3.2",
+        sha256="91eb5088069f4e6edab69e14c4212f6da0192e65695956dc048016a0dab8bcf6",
+        url="https://github.com/neovim/deps/raw/6c0a5213d062f2210fd5fecb9524ac27ca018ef4/opt/libvterm-0.3.2.tar.gz",
+    )
     version("0.3.1", sha256="25a8ad9c15485368dfd0a8a9dca1aec8fea5c27da3fa74ec518d5d3787f0c397")
     version("0.3", sha256="61eb0d6628c52bdf02900dfd4468aa86a1a7125228bab8a67328981887483358")
     version("0.2", sha256="4c5150655438cfb8c57e7bd133041140857eb04defd0e544521c0e469258e105")
```

It is not exactly a mirror but, assuming we can trust `neovim/deps`, it might be enough just copying updates of https://github.com/neovim/neovim/blob/master/cmake.deps/deps.txt where both url and sha256 are listed.

**Comment by G-Ragghianti** [comment]
Yes, thanks for looking into this.  I also attempted to get neovim to build with built-in dependencies, but I had no luck.  I also noticed that libtermkey is hosted on the same problematic site and is also a dep of neovim.  I found that libtermkey is hosted on the neovim github account (but is missing the most recent release), and libvterm is hosted on launchpad.net (seems to be usable, same hashes).
